iib/workers/tasks/opm_operations.py

```diff
@@ -193,6 +193,134 @@ def _serve_cmd_at_port(serve_cmd, cwd, port, max_tries, wait_time):
+
+    :param str base_dir: base directory where index.db file will be located.
+    :param str from_index: index image, from which we should copy index.
+    :return: Returns path to index.db located in base_dir.
+    :rtype: str
+    """
+    index_db_file = os.path.join(base_dir, get_worker_config()['temp_index_db_path'])
+
+    if os.path.exists(index_db_file):
+        log.debug('Temp index.db already exist for %s', from_index)
+        return index_db_file
+
+    log.info('Temp index.db does not exist yet for %s', from_index)
+    if from_index:
+        assert is_image_fbc(from_index), 'Only File-Based Catalog index images are supported'
+        log.info('Using the existing database from %s', from_index)
+        return get_hidden_index_database(from_index, base_dir)
+    else:
+        log.info('Creating empty database file %s', index_db_file)
+        index_db_dir = os.path.dirname(index_db_file)
+        if not os.path.exists(index_db_dir):
+            os.makedirs(index_db_dir, exist_ok=True)
+        with open(index_db_file, 'w'):
+            pass
+
+    return index_db_file
+
+
+def opm_registry_deprecatetruncate(
+    base_dir,
+    index_db,
+    bundles,
+    from_index=None,
+    overwrite_target_index_token=None,
+    container_tool=None,
+):
+    """
+    Deprecate bundles from index.db.
+
+    :param str base_dir: base directory where operation files will be located.
+    :param str index_db: path to index.db used with opm registry deprecatetruncate.
+    :param list bundles: pull specifications of bundles to deprecate.
+    :param str from_index: the pull specification of the container image to parse to determine
+        the registry this token is for.
+    :param str overwrite_target_index_token: the token used for overwriting the input
+        ``from_index`` image. This is required to use ``overwrite_target_index``.
+        The format of the token must be in the format "user:password".
+    :param str container_tool: the container tool to be used to operate on the index image
+    """
+    from iib.workers.tasks.utils import run_cmd, set_registry_token
+
+    log.debug(
+        'Run opm registry deprecatetruncate on database %s and bundles %s',
+        index_db,
+        ' '.join(bundles),
+    )
+
+    cmd = [
+        'opm',
+        'registry',
+        'deprecatetruncate',
+        '--database',
+        index_db,
+        '--bundle-images',
+        ','.join(bundles),
+        '--allow-package-removal',
+    ]
+    if container_tool:
+        cmd.append('--container-tool')
+        cmd.append(container_tool)
+    with set_registry_token(overwrite_target_index_token, from_index):
+        run_cmd(cmd, {'cwd': base_dir}, exc_msg=f'Failed to deprecate the bundles on {index_db}')
+
+
+def deprecate_bundles_fbc(
+    bundles,
+    base_dir,
+    binary_image,
+    from_index,
+    overwrite_target_index_token=None,
+    container_tool=None,
+):
+    """
+    Deprecate the specified bundles from the index image.
```

Comment:
I know it might seem a bit redundant but can you please change this description to something like: "Deprecate the specified bundles from the index image when it's FBC?"

Because this text is exactly the same from `iib.workers.tasks.utils.deprecate_bundles` and I believe some confusion might appear.

Also please consider removing the `deprecate_bundles` from `utils.py` as mentioned in the other comment.

Reply:
Updated docstring, I can not remove different function - see the other comment (response from Tulsi)